documentation/guide_to_branches_and_PRs.md

```diff
@@ -0,0 +1,71 @@
+# Step-by-Step Guide to a Good Branch and Pull Request (PR)
+
+## Creating a New Branch
+
+1. **Always start from the `main` branch.**
+   - Before starting any new task, ensure you are on the latest `main` branch. 
+   - Pull the latest changes to make sure your branch is up to date, preventing conflicts and ensuring you're working with the current version of the codebase.
+
+2. **Create a new branch for each specific task.**
+   - In the Source Control panel, create a new branch with a clear and descriptive name. 
+   - Use a consistent naming convention such as `feature/short-description` or `bugfix/issue-number` to reflect the task you’re working on.
+
+## Keep Branch Scope Focused
+
+3. **Ensure each branch focuses on a single task or issue.**
+   - Each branch should focus on a single task or issue. Avoid including multiple, unrelated changes in one branch. 
+   - If fixing multiple small, closely related issues, they may be grouped into one branch, but limit this to avoid scope creep and ensure easy review.
+
+## Committing Work to Your Branch
+
+4. **Commit frequently and with clear messages.**
+   - Each commit should represent a logical chunk of work. Commit often, but ensure your changes are relevant and cohesive.
+   - Avoid bundling unrelated changes in a single commit. Write clear, descriptive commit messages that explain what has changed (e.g., `Add validation for user email field`).
+
+## Keeping Your Branch Up to Date
+
+5. **Regularly pull changes from `main` to keep your branch up to date.**
+   - If `main` has changed significantly, pull the latest changes into your branch. 
+   - This helps prevent merge conflicts later on and ensures that your branch remains compatible with the rest of the codebase. It’s best to do this frequently rather than waiting until the end of your task.
+
+## Writing a Clear Pull Request (PR)
+
+6. **Ensure the PR focuses on one task.**
+   - Your pull request should be scoped to one feature or bug fix. 
+   - Keeping the PR focused makes the review process faster, easier, and ensures that testing is straightforward.
+
+7. **Reference related GitHub issues.**
+   - In the PR description, reference the GitHub issue number that the PR addresses (e.g., `Fixes #123`). 
```

Comment:
A small point within this document, or having another short doc, on a guide to GitHub Issues might be necessary. e.g. it would set guidelines known by all, especially those new to git that may be unsure about how it should be used.

Reply:
noted and agreed - so just to clarify; also a guide_to_github_issues?